webapp/packages/core-root/src/SessionActivityService.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2024 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+import { ClientActivityService } from '@cloudbeaver/core-client-activity';
+import { Dependency, injectable } from '@cloudbeaver/core-di';
+
+import { SessionResource } from './SessionResource';
+
+export const SESSION_TOUCH_TIME_PERIOD = 1000 * 60;
+
+@injectable()
+export class SessionActivityService extends Dependency {
+  private touchSessionTimer: ReturnType<typeof setTimeout> | null = null;
+
+  constructor(private readonly clientActivityService: ClientActivityService, private readonly sessionResource: SessionResource) {
+    super();
+    this.notifyClientActivity = this.notifyClientActivity.bind(this);
+    this.clientActivityService.onActiveStateChange.addHandler(this.notifyClientActivity);
+  }
+
+  private async notifyClientActivity() {
+    if (this.touchSessionTimer || !this.clientActivityService.isActive) {
+      return;
+    }
+
+    try {
+      await this.sessionResource.touchSession();
+    } catch (e) {
+      console.error('Session touch error', e);
```

Comment:
It is forbidden to import notificationService into core-root, so I logged it with this